frontend/src/scenes/data-pipelines/batch-exports/batchExportConfigurationLogic.tsx

```diff
@@ -566,8 +604,49 @@ export const batchExportConfigurationLogic = kea<batchExportConfigurationLogicTy
                         filters,
                         json_config_file,
                         integration_id,
+                        // Redshift COPY configuration
+                        mode,
+                        authorization_mode,
+                        redshift_s3_bucket,
+                        redshift_s3_key_prefix,
+                        redshift_s3_bucket_region_name,
+                        redshift_s3_bucket_aws_access_key_id,
+                        redshift_s3_bucket_aws_secret_access_key,
+                        redshift_iam_role,
+                        redshift_aws_access_key_id,
+                        redshift_aws_secret_access_key,
                         ...config
                     } = formdata
+
+                    if (destination === 'Redshift') {
+                        if (mode === 'COPY') {
+                            const copyInputs: Record<string, any> = {
+                                s3_bucket: redshift_s3_bucket,
+                                s3_key_prefix: redshift_s3_key_prefix,
+                                region_name: redshift_s3_bucket_region_name,
+                            }
+
+                            if (redshift_iam_role) {
+                                copyInputs.authorization = redshift_iam_role
+                            } else if (redshift_aws_access_key_id && redshift_aws_secret_access_key) {
+                                copyInputs.authorization = {
+                                    aws_access_key_id: redshift_aws_access_key_id,
+                                    aws_secret_access_key: redshift_aws_secret_access_key,
+                                }
+                            }
+
+                            if (redshift_s3_bucket_aws_access_key_id && redshift_s3_bucket_aws_secret_access_key) {
+                                copyInputs.bucket_credentials = {
+                                    aws_access_key_id: redshift_s3_bucket_aws_access_key_id,
+                                    aws_secret_access_key: redshift_s3_bucket_aws_secret_access_key,
+                                }
+                            }
+
+                            config.copy_inputs = copyInputs
```

Comment:
It's not clear to me what would happen if the user selects `COPY` mode but doesn't fill in all the required fields. Perhaps we should add some validation if this doesn't already exist?

Reply:
To be honest, I don't know how to achieve this "conditional validation" as the fields are only required when `COPY` is selected.